src/main/java/com/bootcamp/demo/restapi/BookingsController.java

```
@@ -0,0 +1,174 @@
+import org.springframework.web.bind.annotation.*;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.LinkedHashSet;
+import java.util.Random;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+import java.util.logging.Logger;
+
+
+@RestController
+@RequestMapping(path = "/api/bookings")
+public class BookingsController {
+    private static final ResponseEntity<Object> SUCCESS_RESPONSE = new ResponseEntity<>(HttpStatus.OK);
+    private static final ResponseEntity<Object> FAILURE_RESPONSE = new ResponseEntity<>(HttpStatus.BAD_REQUEST);
+    private static final Logger LOGGER = Logger.getLogger(BookingsController.class.getName());
+    private final BookingService bookingService;
+    private final ScooterService scooterService;
+
+    public BookingsController(BookingService bookingService, ScooterService scooterService) {
+        this.bookingService = bookingService;
+        this.scooterService = scooterService;
+    }
+
+    @PostMapping
+    @ResponseStatus(value = HttpStatus.CREATED)
+    public ResponseEntity<Object> createBooking(@RequestBody final Booking booking) {
+        try {
+            LOGGER.info("Booking created successfully. Update time: " + bookingService.createBooking(booking));
+            return SUCCESS_RESPONSE;
+        } catch (ExecutionException | InterruptedException | IllegalArgumentException e) {
+            LOGGER.info("Unfortunately, booking could not be  created .");
+            return FAILURE_RESPONSE;
+        }
+    }
+
+    @GetMapping("/getBookingsByUserId/{userId}")
+    @ResponseBody
+    public ResponseEntity<LinkedHashSet<Booking>> getBookingsByUserId(@PathVariable(value = "userId") long userId) {
+        LinkedHashSet<Booking> bookingsByUserId = new LinkedHashSet<>();
+        try {
+            LOGGER.info("GET BOOKINGS BY userID - API endpoint invoked");
+
+            bookingsByUserId = bookingService.getBookings(userId);
```

Rather than assigning a value here or returning the empty set on failure, you could inline these.